src/h5web/vis-packs/core/hooks.ts

```diff
@@ -8,24 +8,31 @@ import type { DimensionMapping } from '../../dimension-mapper/models';
 import { getCanvasScale, getDomain } from './utils';
 import AxisSystemContext from './shared/AxisSystemContext';
 import type { AxisScale } from './models';
-import type { HDF5Value } from '../../providers/hdf5-models';
+import type { HDF5Shape, HDF5Type } from '../../providers/hdf5-models';
 import { ProviderContext } from '../../providers/context';
-import type { Dataset } from '../../providers/models';
+import type { Dataset, Value } from '../../providers/models';
 
-export function useDatasetValue(
-  path: string | undefined
-): HDF5Value | undefined {
+export function useDatasetValue<
+  S extends HDF5Shape,
+  T extends HDF5Type,
+  D extends Dataset<S, T> | undefined
+>(dataset: D): D extends Dataset<infer S, infer T> ? Value<S, T> : undefined;
```

Comment:
Same as with `useBaseArray` and `useDataArray`: the conditional return type allows inferring for instance `number[]` instead of `number[] | undefined` (which would have to be followed by `assertDefined` or `as number[]`).

The other way to write it would be to replace the override above with two slightly simpler, but repetitive, overrides:

```ts
export function useDatasetValue<S extends HDF5Shape, T extends HDF5Type>(
  dataset: Dataset<S, T>
): Value<S, T>;

export function useDatasetValue<S extends HDF5Shape, T extends HDF5Type>(
  dataset: Dataset<S, T> | undefined
): Value<S, T> | undefined;
```